Description: moved breakpoint typedef and BPT_* macros from dbg_breakpoints.h into m64p_types.h
Origin: upstream, https://bitbucket.org/richard42/mupen64plus-core/changeset/769d4a908a0c
Author: Richard Goedeken <Richard@fascinationsoftware.com>

---
diff --git a/src/api/m64p_types.h b/src/api/m64p_types.h
index 8a965f81433b39e451bbe0cef847f702ae02c15a..6062d7d6b099bb352868d1e348b06a4f87499438 100644
--- a/src/api/m64p_types.h
+++ b/src/api/m64p_types.h
@@ -233,6 +233,28 @@ typedef enum {
   M64P_BKP_CMD_CHECK
 } m64p_dbg_bkp_command;
 
+#define BREAKPOINTS_MAX_NUMBER  128
+
+#define BPT_FLAG_ENABLED        0x01
+#define BPT_FLAG_CONDITIONAL    0x02
+#define BPT_FLAG_COUNTER        0x04
+#define BPT_FLAG_READ           0x08
+#define BPT_FLAG_WRITE          0x10
+#define BPT_FLAG_EXEC           0x20
+#define BPT_FLAG_LOG            0x40 //Log to the console when this breakpoint hits.
+
+#define BPT_CHECK_FLAG(a, b)  ((a.flags & b) == b)
+#define BPT_SET_FLAG(a, b)    a.flags = (a.flags | b);
+#define BPT_CLEAR_FLAG(a, b)  a.flags = (a.flags & (~b));
+#define BPT_TOGGLE_FLAG(a, b) a.flags = (a.flags ^ b);
+
+typedef struct _breakpoint {
+    unsigned int address; 
+    unsigned int endaddr;
+    unsigned int flags;
+    //unsigned int condition;  //Placeholder for breakpoint condition
+    } breakpoint;
+
 /* ------------------------------------------------- */
 /* Structures and Types for Core Video Extension API */
 /* ------------------------------------------------- */
diff --git a/src/debugger/dbg_breakpoints.h b/src/debugger/dbg_breakpoints.h
index 0a7152c5b65447366a82ea693b0f110a2686eb5a..e6d163fe5d30989794c4c83908cee27af8c740b4 100644
--- a/src/debugger/dbg_breakpoints.h
+++ b/src/debugger/dbg_breakpoints.h
@@ -22,27 +22,7 @@
 #ifndef __BREAKPOINTS_H__
 #define __BREAKPOINTS_H__
 
-#define BREAKPOINTS_MAX_NUMBER  128
-
-#define BPT_FLAG_ENABLED        0x01
-#define BPT_FLAG_CONDITIONAL    0x02
-#define BPT_FLAG_COUNTER        0x04
-#define BPT_FLAG_READ           0x08
-#define BPT_FLAG_WRITE          0x10
-#define BPT_FLAG_EXEC           0x20
-#define BPT_FLAG_LOG            0x40 //Log to the console when this breakpoint hits.
-
-#define BPT_CHECK_FLAG(a, b)  ((a.flags & b) == b)
-#define BPT_SET_FLAG(a, b)    a.flags = (a.flags | b);
-#define BPT_CLEAR_FLAG(a, b)  a.flags = (a.flags & (~b));
-#define BPT_TOGGLE_FLAG(a, b) a.flags = (a.flags ^ b);
-
-typedef struct _breakpoint {
-    uint32 address; 
-    uint32 endaddr;
-    uint32 flags;
-    //uint32 condition;  //Placeholder for breakpoint condition
-    } breakpoint;
+#include "../api/m64p_types.h"
 
 extern int g_NumBreakpoints;
 extern breakpoint g_Breakpoints[];
